components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -321,6 +369,60 @@ public Resource getResourceById(String resourceId) throws ConfigurationManagemen
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Resource getTenantResourceById(int tenantId, String resourceId) throws ConfigurationManagementException {
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        List<ConfigurationRawDataCollector> configurationRawDataCollectors;
+        try {
+            StringBuilder sb = new StringBuilder();
+            sb.append(useCreatedTimeField() ? GET_RESOURCE_BY_ID_MYSQL :
+                    GET_RESOURCE_BY_ID_MYSQL_WITHOUT_CREATED_TIME);
+            sb.append(" AND R.TENANT_ID = ?");
```

Comment:
Instead, shall we define a separate query constant for ` AND R.TENANT_ID = ?`?

Reply:
This was decided during the last code review to avoid the code duplication

Reply:
+1 for that. What I meant is to use a constant for the tenant ID appending query